test/test_threaded.jl

```diff
@@ -47,10 +48,11 @@ Trixi.mpi_isroot() && isdir(outdir) && rm(outdir, recursive=true)
   @testset "StructuredMesh" begin
     @trixi_testset "elixir_advection_restart.jl with waving flag mesh" begin
       @test_trixi_include(joinpath(examples_dir(), "structured_2d_dgsem", "elixir_advection_restart.jl"),
-        l2   = [0.00017274040834067234],
-        linf = [0.0015435741643734513],
+        l2   = [0.00016265538283059892],
+        linf = [0.0015194298895079283],
+        rtol = 3e-7,
```

Comment:
These are the errors I get when running the test serially. It fails in CI without increasing the tolerance. Should I put the threaded errors there instead, or keep the serial errors with a higher tolerance?

Reply:
It needs an even higher tolerance in macOS...

Reply:
Did you check the error of the version producing the restart file?

Reply:
What do you mean?

Reply:
Interesting, the errors before the restart (and also at time step 0 after restarting) are pretty much identical:

Ubuntu after restarting:
```
────────────────────────────────────────────────────────────────────────────────────────────────────
 Simulation running 'LinearScalarAdvectionEquation2D' with DGSEM(polydeg=3)
────────────────────────────────────────────────────────────────────────────────────────────────────
 #timesteps:                  0                run time:       6.00000000e-07 s
 Δt:             1.00000000e+00                time/DOF/rhs!:         NaN s
 sim. time:      1.00000000e+00
 #DOF:                     4096
 #elements:                 256

 Variable:       scalar        
 L2 error:       1.85538599e-04
 Linf error:     1.61677190e-03
 ∑∂S/∂U ⋅ Uₜ :  -3.93688385e-07
────────────────────────────────────────────────────────────────────────────────────────────────────


────────────────────────────────────────────────────────────────────────────────────────────────────
 Simulation running 'LinearScalarAdvectionEquation2D' with DGSEM(polydeg=3)
────────────────────────────────────────────────────────────────────────────────────────────────────
 #timesteps:                 21                run time:       2.11916630e-02 s
 Δt:             2.77991086e-02                time/DOF/rhs!:  3.79310177e-08 s
 sim. time:      2.00000000e+00
 #DOF:                     4096
 #elements:                 256

 Variable:       scalar        
 L2 error:       1.62655384e-04
 Linf error:     1.51943031e-03
 ∑∂S/∂U ⋅ Uₜ :  -3.74186097e-07
────────────────────────────────────────────────────────────────────────────────────────────────────
```
macOS after restarting:
```
────────────────────────────────────────────────────────────────────────────────────────────────────
 Simulation running 'LinearScalarAdvectionEquation2D' with DGSEM(polydeg=3)
────────────────────────────────────────────────────────────────────────────────────────────────────
 #timesteps:                  0                run time:       8.34000000e-07 s
 Δt:             1.00000000e+00                time/DOF/rhs!:         NaN s
 sim. time:      1.00000000e+00
 #DOF:                     4096
 #elements:                 256

 Variable:       scalar        
 L2 error:       1.85538599e-04
 Linf error:     1.61677191e-03
 ∑∂S/∂U ⋅ Uₜ :  -3.93688385e-07
────────────────────────────────────────────────────────────────────────────────────────────────────


────────────────────────────────────────────────────────────────────────────────────────────────────
 Simulation running 'LinearScalarAdvectionEquation2D' with DGSEM(polydeg=3)
────────────────────────────────────────────────────────────────────────────────────────────────────
 #timesteps:                 21                run time:       1.24217080e-02 s
 Δt:             2.77991086e-02                time/DOF/rhs!:  1.28242438e-08 s
 sim. time:      2.00000000e+00
 #DOF:                     4096
 #elements:                 256

 Variable:       scalar        
 L2 error:       1.62655390e-04
 Linf error:     1.51943937e-03
 ∑∂S/∂U ⋅ Uₜ :  -3.74186281e-07
────────────────────────────────────────────────────────────────────────────────────────────────────
```